custom/src/main/java/co/elastic/otel/dynamicconfig/CentralConfig.java

```diff
@@ -0,0 +1,271 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.otel.dynamicconfig;
+
+import co.elastic.opamp.client.CentralConfigurationManager;
+import co.elastic.opamp.client.CentralConfigurationProcessor;
+import co.elastic.otel.logging.AgentLog;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import io.opentelemetry.sdk.trace.SdkTracerProviderBuilder;
+import java.text.MessageFormat;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Logger;
+
+public class CentralConfig {
+  private static final Logger logger = Logger.getLogger(CentralConfig.class.getName());
+
+  static {
+    DynamicConfigurationPropertyChecker.startCheckerThread();
```

Comment:
Do we always want to start this or only if opamp is actually enabled?
If the latter, we should probably move this to the `init` method instead?

Reply:
always, it's primarily for testing though can be used in production